File: 48-rotate-image/rotate-image.cpp
class Solution {
public:
    void rotate(vector<vector<int>>& arr) {
        int n= (int)arr.size();
        for(int i=0;i<n;++i){
            for(int j=i+1;j<n;++j){
                swap(arr[i][j],arr[j][i]);
            }
        }
        for(int i=0;i<n/2;++i){
            for(int j=0;j<n;++j){
                swap(arr[j][i],arr[j][n-1-i]);
            }
        }
    }
};

File: 48-rotate-image/rotate-image.java
class Solution {
    private void swap(int[][] matrix, int row, int col){
        int temp = matrix[row][col];
        matrix[row][col] = matrix[col][row];
        matrix[col][row] = temp;
    }
    private void swap(int[][] matrix, int row, int col1, int col2){
        int temp = matrix[row][col1];
        matrix[row][col1] = matrix[row][col2];
        matrix[row][col2] = temp;
    }
    public void rotate(int[][] matrix) {
        int n = matrix.length;
        if(n<2) return;
        for(int i=1;i<n;i++){
            for(int j=0;j<i;j++){
                swap(matrix, i, j);
            }
        }
        for(int j=0;j<n/2;j++){
            for(int i=0;i<n;i++){
                swap(matrix, i, j, n-1-j);
            }
        }
    }
}